pkg/graph/constructor.go

```diff
@@ -17,14 +17,134 @@ limitations under the License.
 package graph
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	rest "k8s.io/client-go/rest"
+
 	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
 	eventingv1beta3 "knative.dev/eventing/pkg/apis/eventing/v1beta3"
 	messagingv1 "knative.dev/eventing/pkg/apis/messaging/v1"
+	eventingclient "knative.dev/eventing/pkg/client/clientset/versioned"
 	duckv1 "knative.dev/pkg/apis/duck/v1"
 )
 
+type ConstructorConfig struct {
+	RestConfig            rest.Config
+	Namespaces            []string
+	ShouldAddBroker       func(b eventingv1.Broker) bool
+	ShouldAddChannel      func(c messagingv1.Channel) bool
+	ShouldAddSource       func(s duckv1.Source) bool
+	ShouldAddTrigger      func(t eventingv1.Trigger) bool
+	ShouldAddSubscription func(s messagingv1.Subscription) bool
+	ShouldAddEventType    func(et eventingv1beta3.EventType) bool
+}
+
+func ConstructGraph(ctx context.Context, config ConstructorConfig, filterFunc func(obj interface{}) bool) (*Graph, error) {
+	eventingClient, err := eventingclient.NewForConfig(&config.RestConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	g := NewGraph()
+
+	for _, ns := range config.Namespaces {
```

Comment:
I was thinking for this you could just provide `metav1.NamespaceAll` as the only namespace in the config list. Would that work?

Reply:
I think it would...

We might then need add a validation in the code to check if the list of namespaces contain "metav1.NamespaceAll" and if so, it should not have anything else.